src/components/CurrentCityWeather/CurrentCityWeather.js

```diff
@@ -1,23 +1,33 @@
-import { useEffect, useState, useRef  } from "react";
+import { useEffect, useState, useRef } from "react";
 import "./CurrentCityWeather.css";
 import Autocomplete from "react-google-autocomplete";
 import Background from "../../data/BackGroundAccordingToWeather";
 import { Col, Row } from "react-bootstrap";
+import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faMicrophone, faMicrophoneSlash } from '@fortawesome/free-solid-svg-icons'
 
 const CurrentCityWeather = () => {
+
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
   const [city, setCity] = useState("New York City");
   const [results, setResults] = useState(null);
 
-  
+  const {
+    transcript,
+    listening,
+    browserSupportsSpeechRecognition
+  } = useSpeechRecognition();
+
+
   useEffect(() => {
     fetch(
       "https://api.openweathermap.org/data/2.5/weather?q=" +
-        city +
-        "&units=metric" +
-        "&appid=" +
-        process.env.REACT_APP_APIKEY
+      city +
+      "&units=metric" +
+      "&appid=" +
+      process.env.REACT_APP_APIKEY
```

Comment:
I saw this somewhere recently and have started using it since it makes thing a little easier to read:

```js
      "https://api.openweathermap.org/data/2.5/weather?"
      + `q=${city}&`
      + "units=metric&"
      + `appid=${process.env.REACT_APP_APIKEY}`
```